internal/ast/parseoptions.go

```diff
@@ -0,0 +1,162 @@
+package ast
+
+import (
+	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/tspath"
+)
+
+type JSDocParsingMode int
+
+const (
+	JSDocParsingModeParseAll JSDocParsingMode = iota
+	JSDocParsingModeParseNone
+	JSDocParsingModeParseForTypeErrors
+	JSDocParsingModeParseForTypeInfo
+)
+
+type SourceFileParseOptions struct {
+	FileName                       string
+	Path                           tspath.Path
+	CompilerOptions                core.SourceFileAffectingCompilerOptions
+	ExternalModuleIndicatorOptions ExternalModuleIndicatorOptions
+	JSDocParsingMode               JSDocParsingMode
+}
+
+func GetSourceFileAffectingCompilerOptions(fileName string, options *core.CompilerOptions) core.SourceFileAffectingCompilerOptions {
+	// Declaration files are not parsed/bound differently depending on compiler options.
+	if tspath.IsDeclarationFileName(fileName) {
+		return core.SourceFileAffectingCompilerOptions{}
```

Comment:
It turns out that it's entirely safe to not use any options (source affecting, external module indicator) at all for declaration files, so we can now share these ASTs everywhere.